### ISSUE TITLE
Authentication and Tokens

### ISSUE THREAD

**Issue by popnfresh234** [bluewave-labs/bluewave-uptime#41]
I believe we are going to develop a stateless RESTful API, if that's the case we'll likely need some authentication strategies, at least two:

1.  Provide local login (username/password)
2. 3d party OAuth (Google, Facebook, etc)

Are we going to use a library for this like [Passport.js](https://www.passportjs.org/)?  Or roll our own?

I assume we'll issue a token once a user authenticates, will we use [JWT](https://jwt.io/) for this?  Looks like [jsonwebtoken](https://github.com/auth0/node-jsonwebtoken) is still the go to for this.

We can then write some middleware to verify the JWT on protected routes, or just use `Passport.js`.

In summary it might be helpful to:

*  Decide on authentication strategies
* Decide which libraries (if any) to use for Oauth and JWT (if we decide on that standard)

**Comment by Skorpios604** [comment]
I haven't used passport before but I took a look at the link you posted and it seems pretty straight forward and good for our needs. I have used JWT's so I am familiar with them. Yes you are right they are still very popular. I also like the idea of 3rd Party OAuth. Lots of registration pages usually have that option now. Overall I like the proposed ideas and standard.

**Comment by veyselboybay** [comment]
I haven't used passport.js before as well. We should go with JWT since all of us are familiar with it, and we can easily create our own middleware for valid token check. For the third party logins, I think we should not work on them right now, we can implement them in the second iteration.

**Comment by popnfresh234** [comment]
Sounds good to me, it won't be difficult to integrate 3rd party OAuth later on
